### ISSUE TITLE
Add highlighting for manpages?

### ISSUE THREAD

**Issue by LunarLambda** [sharkdp/bat#523]
I would like to use `bat` as my `MANPAGER`. I am not a fan of `most(1)`, my current alternative is using `nvim(1)` with the `:Man!` command.

**Comment by sharkdp** [comment]
Thank you for the feedback! I thought a little bit about this and played around with `nvim`s version, which is really cool! If you set
``` bash
export MANPAGER="nvim -c 'set ft=man' -"
```
you don't need to call `:Man!` manually. If I understand this correctly, they basically have a specialized vim syntax (https://github.com/neovim/neovim/blob/master/runtime/syntax/man.vim) which highlights certain parts of `man`s output like section headings (`PROLOG`, `SYNOPSIS`, `COPYRIGHT`, …).

One thing which is really neat is that they highlight the `SYNOPSIS` part with C syntax if the man page comes from a corresponding section.

I'm not really sure if there is something that we could do even better than `nvim`, but that's definitely something we could look into. I think we might be able to write a specialized Sublime syntax to do the same. We would then simply need to set `MANPAGER="bat -p -l man"`.

**Comment by LunarLambda** [comment]
I have MANPAGER set like that already (`-c Man!` works too on the version I have), which works okay but isn't ideal for me.

Thank you for considering it!

**Comment by sharkdp** [comment]
> which works okay but isn't ideal for me.

What are changes or features you are looking for?

**Comment by LunarLambda** [comment]
I'm used to `less` controls, which `nvim` emulates somewhat when viewing manpages but not perfectly (mostly scrolling is missing), and it's visually a little distracting to me. I'd mostly like the highlighting as a reading aid.

**Comment by sharkdp** [comment]
I've been playing around with this for a little bit. Here is a rather hacky first version.

If somebody wants to try it (or even work on this):

#### Step 1
Save the following syntax as `~/.config/bat/syntaxes/Man.sublime-syntax`.
```yaml
%YAML 1.2
---
# http://www.sublimetext.com/docs/3/syntax.html
name: Man
file_extensions:
  - man
scope: source.man

variables:
  section_heading: '^\S.*$'

contexts:
  main:
    - match: ^
      push: first_line

  first_line:
    - match: '([A-Z0-9]+)(\()([^)]+)(\))'
      captures:
        1: meta.preprocessor
        2: keyword.operator
        3: string.quoted.other
        4: keyword.operator

    - match: '$'
      push: body

  body:
    - match: '^(SYNOPSIS|SYNTAX|SINTASSI|SKŁADNIA|СИНТАКСИС|書式)'
      push: Packages/C/C.sublime-syntax
      scope: markup.heading
      with_prototype:
        - match: '(?={{section_heading}})'
          pop: true

    - match: '^\S.*$'
      scope: markup.heading

    - match: '\b([a-z0-9_]+)(\()([^)]*)(\))'
      captures:
        1: entity.name.function
        2: keyword.operator
        3: constant.numeric
        4: keyword.operator
```

#### Step 2
Run
```
bat cache --build
```

#### Step 3
Set up the pager:
```
export MANPAGER="sh -c 'sed -e s/.\\\\x08//g | bat -l man -p'"
```

#### Step 4
Run `man`. For now, we do not make any decision based on the man section.
```
man select
man fopen
…
```

**Comment by LunarLambda** [comment]
Thanks, I will give it a run tomorrow. :)

**Comment by sharkdp** [comment]
Here is how it looks like (if everything goes well):

![image](https://user-images.githubusercontent.com/4209276/54397543-22e67980-46b7-11e9-8e04-70db86ed51a6.png)

**Comment by LunarLambda** [comment]
Looks good, couple things I would consider:

Name at the top (SELECT) is colored differently from select() further down

select() (function call) shouldn't be highlighted OR be highlighted
differently from page reference (select(3))

I guess that may be ambiguous though, if something of the form f(n) is used
as an example...

Also maybe give page references one uniform color rather than 3 individual
ones (name, parens, section number)

**Comment by SicariusNoctis** [comment]
Slightly missed opportunity for the title, "bat man page highlights".

**Comment by sharkdp** [comment]
I have added my initial version from above to the repository so that everyone can contribute to the further development. See README for instructions.

**Comment by LunarLambda** [comment]
Thank you! 😊

**Comment by sharkdp** [comment]
Released in [bat v0.12](https://github.com/sharkdp/bat/releases/tag/v0.12.0).

**Comment by dufferzafar** [comment]
I tried using bat as my man pager, but it currently lacks the highlight of command line options..

**Comment by sharkdp** [comment]
> it currently lacks the highlight of command line options..

this feature has not been released yet: #718. If you want to help with the next release, please see  #845

**Comment by danisztls** [comment]
Another advantage to neovim is that it is, anecdotally, faster to open.
